### PR TITLE
🌱 Replace `checker.User` with `clients.User`

### PR DESCRIPTION
**Pull request by azeemshaikh38**
#### What kind of change does this PR introduce?
Part of #1943 
(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
NONE

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

File: checker/raw_result.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ type CodeReviewData struct {
 
 // ContributorsData represents contributor information.
 type ContributorsData struct {
-	Users []User
+	Users []clients.Contributor
```
Comment (laurentsimon) [inline checker/raw_result.go:68]: anything wrong with a contributor being a user? They are the same, some fields remain `nil`

Comment (azeemshaikh38) [reply checker/raw_result.go:68]: I didn't want to attempt combining `clients.Contributor` and `clients.User` struct in this PR. If you think removing `clients.Contributor` in favor of `clients.User` makes more sense, I can attempt that in a different PR.

Comment (laurentsimon) [reply checker/raw_result.go:68]: gocha. Let's try in a follow-up PR then. Unless you have reservation. Wdut?

Comment (azeemshaikh38) [reply checker/raw_result.go:68]: Nope, combining the 2 structs makes sense to me.

**Pull request merged**